zip-0227.rst

```diff
@@ -9,7 +9,445 @@
+
+Issuance Bundle
+---------------
+
+An issuance bundle, `IssueBundle`, is the aggregate of all the issuance-related information. Specifically, contains all the issuance actions and the issuer signature on the transaction SIGHASH that validates the issuance itself. It contains the following fields:
+
+- :math:`\mathsf{ik}`: the issuance validating key, that allows the validators to verify that the :math:`\mathsf{AssetId}` is properly associated with the issuer.
+- `actions`: an array of issuance actions, of type `IssueAction`.
+- `issueAuthSig`: the signature of the transaction SIGHASH, signed by the issuance authorizing key, :math:`\mathsf{isk}`, that validates the issuance .
+
+The issuance bundle is then added within the transaction format as a new bundle. That is, issuance requires the addition of the following information to the transaction format [#protocol-transactionstructure]_.
+
+======= ==================== ========================== =========================================================================
+Bytes              Name               Data Type                  Description
+======= ==================== ========================== =========================================================================
+Varies  nIssueActions        compactSize                The number of issuance actions in the bundle
+Varies  vIssueActions        IssueAction[nIssueActions] A sequence of issuance actions descriptions
+32      :math:`\mathsf{ik}`  byte[32]                   The issuance validating key of the issuer, used to validate the signature
+64      issueAuthSig         byte[64]                   The signature of the transaction SIGHASH, signed by the issuer
+======= ==================== ========================== =========================================================================
+
+Issuance Protocol
+-----------------
+The issuer program performs the following operations
+
+For all actions `IssueAction`:
+
+- encode :math:`\mathsf{asset\_desc}` as a UTF-8 byte string of size up to 512.
+- compute :math:`\mathsf{AssetDigest}` from the issuance validating key :math:`\mathsf{ik}` and :math:`\mathsf{asset\_desc}` as decribed in the `Asset Identifier`_ section.
+- compute :math:`\mathsf{AssetBase^{Protocol}}` from :math:`\mathsf{AssetDigest}` as decribed in the `Asset Identifier`_ section.
+- set the ``finalize`` boolean as desired (if more more issuance actions are to be created for this Asset Identifier, set ``finalize = 0``, otherwise set ``finalize = 1``)
+- For each recipient :math:`i`:
+
+    - generate a ZSA output note that includes the Asset Base. For an Orchard-based ZSA note this is :math:`\mathsf{note}_i = (\mathsf{d}_i, \mathsf{pk}_{\mathsf{d},i}, \mathsf{v}_i, \rho_i, \psi_i, \mathsf{AssetBase^{Orchard}}, \mathsf{rcm}_i)\!`.
```

Comment:
Do the notes in the issuance actions encode \psi or rseed? This line seems to imply psi is explicitly encoded, but rseed is needed in the SigHash section below to compute the transaction hash.

Reply:
Yes, that's right, we were to include `rseed` there, not `psi`. Thanks for pointing it out, we've updated it.